src/xdShare.sol

```diff
@@ -53,6 +61,74 @@ contract xdShare is Ownable, xERC4626 {
         return address(underlyingDShare);
     }
 
+    /**
+     * @dev Locks the contract to prevent deposit and withdrawal operations.
+     * Can only be called by the owner of the contract.
+     */
+    function lock() public onlyOwner {
+        isLocked = true;
+        emit VaultLocked();
+    }
+
+    /**
+     * @dev Unlocks the contract to allow deposit and withdrawal operations.
+     * Can only be called by the owner of the contract.
+     */
+    function unlock() public onlyOwner {
+        isLocked = false;
+        emit VaultUnlocked();
+    }
+
+    /**
+     * @dev Allows a user to deposit assets into the contract.
+     * Reverts if the contract is locked with a "DepositPaused" error.
+     * @param assets The amount of assets to deposit.
+     * @param to The address to credit the deposit to.
+     * @return shares The amount of shares received in exchange for the deposit.
+     */
+    function deposit(uint256 assets, address to) public virtual override returns (uint256 shares) {
```

Comment:
Override `_deposit` and `_withdraw` rather than deposit, mint, redeem, and withdraw.